cheatsheets/Java_Security_Cheat_Sheet.md

```diff
@@ -598,20 +602,24 @@ class AesGcmSimple {
 
 </details>
 
-#### Symmetric example using built in JCA/JCE classes
+_**End of code snippet.**_
+
+#### Symmetric example using built-in JCA/JCE classes
 
-If you absolutely cannot use a separate library, it is still possible to use the built JCA/JCE classes but it is strongly recommended to have a cryptography expert review the full design and code.
+If you absolutely cannot use a separate library, it is still possible to use the built JCA/JCE classes but it is strongly recommended to have a cryptography expert review the full design and code, as even the most trivial error can severely weaken your encryption.
 
 The following code snippet shows an example of using AES-GCM to perform encryption/decryption of data.
 
-A few contraints/pitfalls with this code:
+A few constraints/pitfalls with this code:
 
 - It does not take into account key rotation or management which is a whole topic in itself.
 - It is important to use a different nonce for every encryption operation, even if the same key is used.

```

Comment:
(We could emphasis the fact that 'nonce' is short for 'number used once'.) But my critique of this bullet is not that but the wording "_even_ if the same key is used". It is _**critical**_ that is has to be different when a key is reused. The whole point is that the same key / IV pair (where here, nonce is your IV) must always be unique for all time (or as long as any adversary might be collecting your ciphertexts). Otherwise it can let to decrypting some of those collected ciphertexts later.

**Strong suggestion**: Change 'even' to 'especially'.

Reply:
Addressed in #1160